handler/crear.go

```diff
@@ -0,0 +1,362 @@
+	t.Bot.Send(tgMessage)
+
+	return nil
+}
+
+
+func (t *Crear) welcomeUser(message *tgbotapi.Message) error {
+	msg := tgbotapi.NewMessage(message.Chat.ID, "Hello "+ message.From.FirstName + " " + message.From.LastName + "!")
+	msg.ReplyMarkup = tgbotapi.NewRemoveKeyboard(false)
+	t.Bot.Send(msg)
+
+	t.sendMsg(message.Chat.ID, "To Start a new log, type /" +cmdNewLog+ " and then forward the messages you want to add. Once finished, type /" + cmdEndLog)
+
+	return nil
+}
+
+func (t *Crear) matchCommand(command string, message *tgbotapi.Message) (bool, error) {
+	if message == nil {
+		return false, nil
+	}
+	return regexp.MatchString("^" + command + "", message.Command())
+}
+
+func (t *Crear) hasInlineKeyboardReply(update tgbotapi.Update) (bool, error) {
+	return update.CallbackQuery != nil, nil
+}
+
+func (t *Crear) reset(message *tgbotapi.Message) (bool, error) {
+	delete(t.ActiveLogs, message.From.ID)
+
+	tmp := tgbotapi.NewMessage(message.Chat.ID, "Operation cancelled. Back to square 1")
+	tmp.ReplyMarkup = tgbotapi.NewRemoveKeyboard(false)
+	t.Bot.Send(tmp)
+
+	return true, nil
+}
+
+
+func (t *Crear) startLogFromMessage(message *tgbotapi.Message) error {
+	log.Println("Starting a new Log for UserID ", message.From.ID)
+
+	if t.userHasLogInProgress(message) {
+		delete(t.ActiveLogs, message.From.ID)
+
+		t.sendMsg(message.Chat.ID, "You already had a log in progress. Too bad. It has been discarded")
+	}
+
+	t.ActiveLogs[message.From.ID] = &pinchito.PlogData{}
+
+	t.sendMsg(message.Chat.ID, "Ready to start creating a new Log.\nForward the messages you want to add. Once finished, type /" + cmdEndLog)
+
+	return nil
+}
+
+func (t *Crear) endLogFromMessage(message *tgbotapi.Message) error {
+	log.Println("Ending existing log for UserID ", message.From.ID)
+
+	pinLog := t.ActiveLogs[message.From.ID]
+	if pinLog == nil {
+		t.sendMsg(message.Chat.ID, "I haven't found any active log. Ignoring you")
+		return nil
+	}
+
+	if len(pinLog.Text) == 0 {
+		t.sendMsg(message.Chat.ID, "Why do you want to create an empty log? Try harder")
+		return nil
+	}
+
+	autor, err := pinchito.GetUserFromTelegramUsername(message.From.UserName)
+	if err != nil {
+		return err
+	}
+	pinLog.Autor = autor.PinId
+	pinLog.Data = time.Now().Unix()
+	pinLog.Protagonista = protagonistIdPending
+
+	t.askForTitle(message)
+
+	return nil
+}
+
+func (t *Crear) appendMessageToLog(message *tgbotapi.Message) error {
+	if message == nil {
+		return nil
+	}
+
+	pinLog := t.ActiveLogs[message.From.ID]
+	if pinLog == nil {
+		return nil
+	}
+
+	line := pinmessage.BuildNewLogLine(message)
+	if len(line) > 0 {
+		pinLog.Text += line
+	}
+	// TODO Handle other types of messages
+	// (Audio, img, ...)
+
+	return nil
+}
+
+func (t *Crear) handleKeyboardReply(update tgbotapi.Update) error {
+	host := os.Getenv("PINCHITO_HOST")
```

Comment:
Can we move this into an attribute of `Crear` and inject it in when we create the handler?